### DISCUSSION TITLE
Generating responses for testing clients

### DISCUSSION THREAD

**Discussion by ryneeverett** [schemathesis/schemathesis#1613] [Ideas]
it seems that this project is all about generating queries in order to test servers. The openapi and graphql specifications specify response types too, so is there any reason we couldn't do the reverse as well? Has anybody looked into this or is there any prior art?

**Comment by Stranger6667** [comment]
Yep, this should be relatively easy. E.g. `hypothesis-jsonschema` can be used to generate such samples directly (i.e. extracting `schema` from responses and passing them to the `hypothesis_jsonschema.from_schema()` function), but I don't see a clear fit in Schemathesis for such a feature.
